### PR TITLE
Use RDS Aurora for the next stack

### PR DESCRIPTION
**Pull request by bjpirt**

### CODE REVIEW

File: shared_terraform/stack_infra_baseline/main.tf [hunk 1]
```diff
@@ -91,7 +91,8 @@ module "route-53" {
   vpc_id               = data.terraform_remote_state.base_infra.outputs.vpc.vpc_id
   tags                 = module.label.tags
   was_internal_lb_name = [module.was_ecs_task.was_dns_lb]
-  db_internal_ip       = [module.db2.internal_ip]
+  db_endpoint          = [module.db2.internal_ip]
```
Comment (brettminnie) [inline shared_terraform/stack_infra_baseline/main.tf:94]: You renamed the module to db, so db2 won't work here

File: shared_terraform/stack_infra_next/main.tf [hunk 2]
```diff
@@ -18,26 +18,18 @@ module "label" {
...
-  vpc_endpoint_ssm_sg_id        = data.terraform_remote_state.base_infra.outputs.vpc.vpc_endpoint_ssm_sg_id
-  setup_script_bucket_name      = data.terraform_remote_state.base_infra.outputs.ec2.setup_script_bucket_name
-  allow_access_to_bucket_arns = [
-    data.terraform_remote_state.base_infra.outputs.ec2.setup_scripts_bucket_arn,
-    data.terraform_remote_state.base_infra.outputs.ec2.installers_bucket_arn
-  ]
-
-  tags = module.label.tags
+  rds_master_username           = "bichard"
+  rds_database_name             = "bichard"
```
Comment (brettminnie) [inline shared_terraform/stack_infra_next/main.tf:28]: Do we not want to set these as vars if we're feeding them into the module

File: terraform/stack_infra_next/main.tf [hunk 3]
```diff
@@ -71,26 +71,18 @@ data "terraform_remote_state" "account_resources" {
...
-  vpc_endpoint_s3_pl_id         = data.terraform_remote_state.base_infra.outputs.vpc.vpc_endpoint_s3_prefix_list_id
-  vpc_endpoint_ssm_sg_id        = data.terraform_remote_state.base_infra.outputs.vpc.vpc_endpoint_ssm_sg_id
-  setup_script_bucket_name      = data.terraform_remote_state.base_infra.outputs.ec2.setup_script_bucket_name
-  allow_access_to_bucket_arns = [
-    data.terraform_remote_state.base_infra.outputs.ec2.setup_scripts_bucket_arn,
-    data.terraform_remote_state.base_infra.outputs.ec2.installers_bucket_arn
-  ]
-
-  tags = module.label.tags
+  rds_master_username           = "bichard"
```
Comment (brettminnie) [inline terraform/stack_infra_next/main.tf:80]: Same as above regarding vars

File: terraform_modules/aurora/main.tf [hunk 4]
```diff
@@ -0,0 +1,84 @@
+resource "random_password" "db" {
+  length  = 24
+  special = true
```
Comment (brettminnie) [inline terraform_modules/aurora/main.tf:3]: Watch this. Postgres can complain with special chars in the connection string 

See https://jasonaowen.net/blog/2017/Feb/09/aws-postgresql-rds-passwords/

File: terraform_modules/aurora/main.tf [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+resource "aws_ssm_parameter" "db_password" {
+  name      = "/${var.environment_name}/rds/db/password"
+  type      = "SecureString"
+  value     = random_password.db.result
+  overwrite = true
+  tags      = var.tags
+}
+
+resource "aws_security_group" "db" {
+  name_prefix = var.environment_name
```
Comment (brettminnie) [inline terraform_modules/aurora/main.tf:15]: Please add a description to the SG

File: terraform_modules/aurora/variables.tf [hunk 6]
```diff
@@ -0,0 +1,40 @@
+variable "tags" {
+  description = "The tags for the resources"
+  type = map(string)
+}
+
+variable "environment_name" {
+  description = "The name of the environment"
```
Comment (brettminnie) [inline terraform_modules/aurora/variables.tf:7]: Add a type

File: terraform_modules/aurora/variables.tf [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+variable "vpc_id" {
```
Comment (brettminnie) [inline terraform_modules/aurora/variables.tf:10]: Add a type and with the vars below

**Review by brettminnie — CHANGES_REQUESTED** [review]

File: shared_terraform/stack_infra_baseline/main.tf (hunk 1 above)
Comment (bjpirt) [reply shared_terraform/stack_infra_baseline/main.tf:94]: This one still has db2 as the module

File: shared_terraform/stack_infra_next/main.tf (hunk 2 above)
Comment (bjpirt) [reply shared_terraform/stack_infra_next/main.tf:28]: They are set as vars in the Aurora module, do they need setting elsewhere?

File: terraform/stack_infra_next/main.tf (hunk 3 above)
Comment (bjpirt) [reply terraform/stack_infra_next/main.tf:80]: likewise

File: terraform_modules/aurora/main.tf (hunk 4 above)
Comment (bjpirt) [reply terraform_modules/aurora/main.tf:3]: We're not using a connection string, this gets configured as an env var on the WAS container and then configured into WAS at startup.

File: terraform_modules/aurora/main.tf (hunk 5 above)
Comment (bjpirt) [reply terraform_modules/aurora/main.tf:15]: Have added types and descriptions for all vars now

File: shared_terraform/stack_infra_next/main.tf (hunk 2 above)
Comment (brettminnie) [reply shared_terraform/stack_infra_next/main.tf:28]: I was referring to having the option of passing them into this terraform from the outside world with these values as defaults

**Review by brettminnie — APPROVED** [review]

**Pull request merged**